### PR TITLE
Remove ewoms register param macro

### PR DESCRIPTION
**Pull request by akva2**
Make things more DRY by using reading parameter type from struct definition and by deriving the parameter name from the struct name. The latter can optionally be overridden by having a name member in the struct.

### CODE REVIEW

**Comment by akva2** [comment]
jenkins build this opm-simulators=5272 please

**Comment by akva2** [comment]
jenkins build this opm-simulators=5272 please

**Review by bska — COMMENTED** [review]
I love the simplification.  Just a couple of missing include statements and then this is good to go I think.

File: opm/models/utils/parametersystem.hh [hunk 1]
```diff
@@ -1114,14 +1081,36 @@ bool isSet(const char* paramName, bool errorIfNotRegistered = true)
...
+    if (!ParamsMeta::registrationOpen()) {
         throw std::logic_error("Parameter registration was already closed before "
-                               "the parameter '"+std::string(paramName)+"' was registered.");
+                               "the parameter '" + paramName + "' was registered.");
+    }
 
+    const auto defaultValue = getPropValue<TypeTag, Param>();
+    using ParamType = std::conditional_t<std::is_same_v<decltype(defaultValue),
+                                                        const char* const>, std::string,
+                                         std::remove_const_t<decltype(defaultValue)>>;
```
[the remark this answers is not in the document]
Comment (akva2) [reply opm/models/utils/parametersystem.hh:1113]: aye.

**Comment by akva2** [comment]
jenkins build this opm-simulators=5272 please

**Comment by akva2** [comment]
hrm, the branch is updated but github is not happy right now it seems.

**Comment by bska** [comment]
> the branch is updated but github is not happy right now it seems.

Yeah...  Just in case you're not aware, [githubstatus.com](https://www.githubstatus.com/) is reporting severe service degradation across the site at the minute.

**Review by bska — APPROVED** [review]
> > the branch is updated but github is not happy right now it seems.
> 
> Yeah... Just in case you're not aware, [githubstatus.com](https://www.githubstatus.com/) is reporting severe service degradation across the site at the minute.

In any case, I've checked the PR branch manually on my local machine and this looks good to me now.  I'll merge this and its downstream counterpart into the master branch.

**Comment by bska** [comment]
> I'll merge this and its downstream counterpart into the master branch.

Okay, so that didn't take.  I guess we'll have to wait for GitHub to recover before we can actually merge this.

**Pull request closed without merging**